tools/serverpod_cli/lib/src/generator/dart/library_generators/util/model_filter_util.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:path/path.dart' as p;
+
+import 'package:serverpod_cli/src/analyzer/models/definitions.dart';
+import 'package:serverpod_cli/src/config/config.dart';
+
+List<ClassDefinition> getSealedClasses(
+  List<SerializableModelDefinition> models,
+) {
+  return models
+      .whereType<ClassDefinition>()
+      .where((element) => element.sealedTopNode == element)
+      .toList();
+}
+
+List<List<ClassDefinition>> getSealedHierarchies(
```

Comment:
Suggestion: Add an `abstract final class` around these helper methods to capture the intention of them.